swerve100/src/main/java/frc/robot/subsystems/SwerveDriveSubsystem.java

```diff
@@ -201,15 +201,17 @@ public SwerveDriveSubsystem(DriverStation.Alliance alliance, double currentLimit
 
                 xController = new PIDController(
                         0.15, // kP
-                        0.0, // kI
+                        0, // kI
                         0.0); // kD
-                xController.setTolerance(0.2);
+                xController.setTolerance(0.01);
+                xController.setIntegratorRange(-0.5, 0.5);
 
                 yController = new PIDController(
                         0.15, // kP
-                        0.0, // kI
+                        0, // kI
                         0.0); // kD
-                yController.setTolerance(0.2);
+                yController.setTolerance(0.01);
+                yController.setIntegratorRange(-0.5, 0.5);
```

Comment:
if kI is zero then the integrator range doesn't do anything.   is something missing here?